packages/edit-site/src/components/left-sidebar/navigation-panel/template-switcher/template-preview.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * WordPress dependencies
+ */
+import { parse } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+import { BlockPreview } from '@wordpress/block-editor';
+import { useMemo } from '@wordpress/element';
+
+function TemplatePreview( { item } ) {
+	const template = useSelect(
+		( select ) => {
+			return select( 'core' ).getEntityRecord(
+				'postType',
+				item.type === 'template' ? 'wp_template' : 'wp_template_part',
+				item.id
+			);
+		},
+		[ item ]
+	);
+	const blocks = useMemo(
+		() => ( template ? parse( template?.content?.raw || '' ) : [] ),
+		[ template ]
```

Comment:
`template` is an object, are we sure this `useMemo` is only updating when `template` changes instead of every render?
Same goes for the `item` dependency of the `useSelect` above.

Do you think it would make more sense to destructure the item ID and use that as dependency?
e.g.
```jsx
function Template Preview( { item } ) {
  const { id: itemId } = item;

  const template = useSelect( () => {}, [ itemId ] );
  const blocks = useMemo( () => {}, [ itemId ] );
}

Reply:
How about using `template?.content?.raw || ''` as a dependency? Probably `itemId` is safe too.